packages/@okta/vuepress-theme-prose/util/attribution/attribution.js

```diff
@@ -0,0 +1,269 @@
+import moment from "moment-timezone";
+import { deleteCookie, getCookie, setCookie } from "./cookies";
+
+const acceptedParams = [
+  "utm_campaign",
+  "utm_content",
+  "utm_medium",
+  "utm_source",
+  "utm_term"
+];
+
+const allParams = [
+  "utm_campaign",
+  "utm_content",
+  "utm_date",
+  "utm_medium",
+  "utm_page",
+  "utm_source",
+  "utm_term"
+];
+
+let isAttached = false;
+
+function filterParams(params) {
+  if (typeof params == "string") {
+    params = queryString.parse(params);
+
+    params = Object.keys(params)
+      .filter(key => acceptedParams.indexOf(key) !== -1)
+      .reduce((object, key) => {
+        object[key] = params[key];
+        return object;
+      }, {});
+  }
+
+  params = Object.keys(params)
+    .filter(key => allParams.indexOf(key) !== -1)
+    .reduce((object, key) => {
+      object[key] = params[key];
+      return object;
+    }, {});
+
+  params = Object.assign(
+    {
+      utm_page: location.pathname,
+      utm_date: moment(moment.utc())
+        .tz("America/Los_Angeles")
+        .format("MM/DD/YYYY")
+    },
+    params
+  );
+
+  return params;
+}
+
+/**
+ * onAttach function.
+ *
+ * @access public
+ * @return void
+ */
+function onAttach() {
+  if (isAttached == false) {
+    isAttached = true;
+    setAttribution();
+  }
+}
+
+/**
+ * setAttribution function.
+ *
+ * @access public
+ * @return void
+ */
+function setAttribution() {
+  const params = filterParams(location.search);
+
+  if (Object.keys(params).length) {
+    deleteCookie("attribution");
+    setCookie("attribution", params);
+
+    if (!getCookie("session_attribution")) {
+      setCookie("session_attribution", params);
+    }
+
+    if (!getCookie("original_attribution")) {
+      setCookie("original_attribution", params, { expires: 365 });
+    }
+  } else {
+    deleteCookie("attribution");
+    setCookie("attribution", {});
+
+    if (!getCookie("session_attribution")) {
+      setCookie("session_attribution", {});
+    }
+
+    if (!getCookie("original_attribution")) {
+      setCookie("original_attribution", {}, { expires: 365 });
+    }
+  }
+}
+
+/**
+ * getAttribution function.
+ *
+ * @access public
+ * @return void
+ */
+
+function getAttribution() {
+  return {
+    page: filterParams(getCookie("attribution", {})),
+    original: filterParams(getCookie("original_attribution", {})),
+    session: filterParams(getCookie("session_attribution", {}))
+  };
+}
+
+function setFieldAttribution($form, key, value = {}) {
+  let $fields, fieldExistsInForm;
+
+  if ("page" in value) {
+    $fields = [].concat(
+      [].slice.call(document.querySelectorAll(`[name=${key}]`)),
+      [].slice.call(document.querySelectorAll(`[name=${key}__c]`))
+    );
+
+    fieldExistsInForm =
+      $form &&
+      (!!$form.querySelectorAll(`[name=${key}]`).length ||
+        !!$form.querySelectorAll(`[name=${key}__c]`).length);
+
+    for (let i = 0; i < $fields.length; i++) {
+      let $el = $fields[i];
```

Comment:
Same question about if we have `forEach` here and in other locations where we have a `for` loop in this file.